src/actions/index.js

```diff
@@ -15,35 +14,68 @@ export function receiveLocation(newLocation) {
   };
 }
 
+export function displaySuccess() {
+  return {
+    type: DISPLAY_SUCCESS,
+    message: 'Request succeeded',
+    color: 'green',
+  };
+}
+
+export function displayError(message) {
+  return {
+    type: DISPLAY_ERROR,
+    message,
+    color: 'red',
+  };
+}
+
+export function sendMessage() {
+  return {
+    type: SENDING_MESSAGE,
+    message: 'Sending request...',
+    color: 'black',
+  };
+}
+
 export function fetchLocation(location) {
   let name;
 
   return dispatch => {
+    dispatch(sendMessage());
 
     return geocodeGoogle(location)
       .then(res => {
-        name = res.results[0].formatted_address;
+        if (res.status === 'OK') {
+          name = res.results[0].formatted_address;
 
-        const lat = res.results[0].geometry.location.lat;
-        const lng = res.results[0].geometry.location.lng;
+          const lat = res.results[0].geometry.location.lat;
+          const lng = res.results[0].geometry.location.lng;
 
-        return { lat, lng };
+          return { lat, lng };
+        } else {
+          dispatch(displayError('Invalid location'));
+        }
       })
       .then(coords => breezoMeter(coords))
       .then(res => {
-        const newLocation = {
-          date: res.datetime,
-          name,
-          aqi: res.breezometer_aqi,
-          description: res.breezometer_description,
-          color: res.breezometer_color,
-        };
-        dispatch(receiveLocation(newLocation));
-
-        dispatch(saveToHistory(newLocation));
+        if (res.data_valid) {
+          const newLocation = {
+            date: res.datetime,
+            name: name,
+            aqi: res.breezometer_aqi,
+            description: res.breezometer_description,
+            color: res.breezometer_color
+          };
+          dispatch(receiveLocation(newLocation));
+          dispatch(displaySuccess());
+          saveToHistory(newLocation);
+        } else if (res.data_valid === false) {
```

Comment:
stylistically I would put this first; "if the data is not valid, throw an error and bail out." Then proceed with the normal code flow.